tests/test_api.py

```diff
@@ -151,10 +167,65 @@ def test_not_found_exception(live_api_client: IpumsApiClient):
     )
 
 
+def test_not_submitted_exception():
+    extract = UsaExtract(
+        ["us2012b"],
+        ["AGE", "SEX"],
+    )
+    with pytest.raises(IpumsExtractNotSubmitted) as exc_info:
+        dct = extract_to_dict(extract)
+    assert exc_info.value.args[0] == (
+        "Extract has not been submitted and so has no json response"
+    )
+
+
 @pytest.mark.integration
 def test_extract_was_purged(live_api_client: IpumsApiClient):
     """
     test extract_was_purged() method
     """
     was_purged = live_api_client.extract_was_purged(extract="1", collection="usa")
     assert was_purged == True
+
+
+def test_extract_from_dict(fixtures_path: Path):
+    with open(str(fixtures_path / "example_extract.yml")) as infile:
+        extract = extract_from_dict(yaml.safe_load(infile))
+        infile.close()
+        for item in extract:
+            assert item.collection == "usa"
+            assert item.samples == ["us2012b"]
+            assert item.variables == ["AGE", "SEX", "RACE"]
+
+    with open(str(fixtures_path / "example_extract.json")) as infile:
+        extract = extract_from_dict(json.load(infile))
+        infile.close()
+        for item in extract:
+            assert item.collection == "usa"
+            assert item.samples == ["us2012b"]
+            assert item.variables == ["AGE", "SEX", "RACE"]
+
+
+def test_extract_to_dict(fixtures_path: Path):
+    # reconstitute the extract object from pickle
+    with open(str(fixtures_path / "usa_extract_obj.pkl"), "rb") as infile:
+        extract = pickle.load(infile)
```

Comment:
I forget when the pickles came in. One thing about pickles is that they get a little funny because the pickle format changes sometimes. We probably should specify the pickle version somewhere....

Reply:
Upon reading this page and also rereading the documentation for `pickle.load`, I retract my comment. Apparently the pickle version is detected automatically and I generally trust the Python core devs to keep their word! Sorry about that!